src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +73,1195 @@ ActorInit En_Slime_InitVars = {
+        }
+        EnSlime_SetupWaitForRevive(this);
+    }
+    this->actor.scale.y = this->actor.scale.x;
+    this->actor.scale.z = this->actor.scale.x;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F180.s")
+/**
+ * Calculate a new "current" position value that is a multiple of 60
+ * away from the "home" position.
+ * This is for use by blue chuchus frozen into an ice block so that
+ * their position is a multiple of the ice block size.
+ */
+f32 EnSlime_SnapIceBlockPosition(f32 currentPosition, f32 homePosition) {
+    s32 dist = currentPosition - homePosition;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F1A4.s")
+    if (dist > 0) {
+        dist += 30;
+    } else {
+        dist -= 30;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F354.s")
+    return ((dist / 60) * 60.0f) + homePosition;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Slime/func_80A2F418.s")
+/**
+ * Prepare actor for ice block to spawn around it.
+ * The actor will stop moving, have AC_ON cleared,
+ * and target position will be snapped to
+ * a multiple of the size of an ice block.
+ */
+void EnSlime_SetupSpawnIceBlock(EnSlime* this) {
+    this->collider.base.acFlags &= ~AC_ON;
+    this->actor.flags &= ~ACTOR_FLAG_1;
+    this->drawDmgEffAlpha = 0.0f;
+    this->actor.speed = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->iceBlockSnapPos.x = EnSlime_SnapIceBlockPosition(this->actor.world.pos.x, this->actor.home.pos.x);
+    this->iceBlockSnapPos.z = EnSlime_SnapIceBlockPosition(this->actor.world.pos.z, this->actor.home.pos.z);
+    Actor_SetColorFilter(&this->actor, 0x8000, 0x80FF, 0x2000, 10);
+    this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+    this->iceBlockTimer = 254;
+    this->actionFunc = EnSlime_SpawnIceBlock;
+}
+
+/**
+ * Attempt to move the chuchu to snapped position and
+ * spawn an ice block actor as a child of the chuchu actor.
+ * If ice block spawn fails, chuchu returns to idle.
+ */
+void EnSlime_SpawnIceBlock(EnSlime* this, PlayState* play) {
+    s32 stepCheck; // bool
+
+    stepCheck = Math_StepToF(&this->actor.world.pos.x, this->iceBlockSnapPos.x, 10.0f);
+    stepCheck &= Math_StepToF(&this->actor.world.pos.z, this->iceBlockSnapPos.z, 10.0f);
+
+    this->actor.colorFilterTimer = 10;
+    if (stepCheck) {
+        this->actor.child =
+            Actor_SpawnAsChild(&play->actorCtx, &this->actor, play, ACTOR_OBJ_ICEBLOCK, this->actor.world.pos.x,
+                               this->actor.world.pos.y + 30.0f, this->actor.world.pos.z, 0, 0, 0, 0);
+        if (this->actor.child != NULL) {
+            this->actor.child->cutscene = this->actor.cutscene;
+            EnSlime_SetupIceBlock(this);
+        } else {
+            this->actor.colorFilterTimer = 0;
+            this->collider.base.acFlags |= AC_ON;
+            this->iceBlockTimer = 255;
```

Comment:
If you make 255 a `#define`d number like I talked about above, make sure to update this too.